contracts/src/v0.8/automation/dev/v2_3/AutomationRegistry2_3.sol

```diff
@@ -250,14 +250,18 @@ contract AutomationRegistry2_3 is AutomationRegistryBase2_3, OCR2Abstract, Chain
     uint64 offchainConfigVersion,
     bytes memory offchainConfig
   ) external override {
-    setConfigTypeSafe(
-      signers,
-      transmitters,
-      f,
-      abi.decode(onchainConfigBytes, (OnchainConfig)),
-      offchainConfigVersion,
-      offchainConfig
+    (OnchainConfig memory config, address[] memory billingTokens, BillingConfig[] memory billingConfigs) = abi.decode(
+      onchainConfigBytes,
+      (OnchainConfig, address[], BillingConfig[])
     );
+    if (billingTokens.length != billingConfigs.length) revert IncorrectBillingTokenConfigLength();
+
+    setConfigTypeSafe(signers, transmitters, f, config, offchainConfigVersion, offchainConfig);
+
+    // set billing configs
+    for (uint256 idx = 0; idx < billingTokens.length; idx++) {
+      setBillingConfig(billingTokens[idx], billingConfigs[idx]);
+    }
```

Comment:
Have this logic here in setConfig instead of setConfigTypeSafe to avoid stack too deep issue.